tests/functional/analyze/test_analyze.py

```diff
@@ -42,6 +42,41 @@ def tearDown(self):
         """Restore environment after tests have ran."""
         os.chdir(self.__old_pwd)
 
+    def test_capture_analysis_output(self):
+        """
+        Test if reports/success/<output_file>.[stdout,stderr] files are created
+        """
+        build_json = os.path.join(self.test_workspace, "build_success.json")
+        reports_dir = os.path.join(self.test_workspace, "reports")
+        success_dir = os.path.join(reports_dir, "success")
+        analyze_cmd = [self._codechecker_cmd, "analyze", build_json,
+                       "--analyzers", "clangsa", "-o", reports_dir,
+                       "--capture-analysis-output"]
+
+        source_file = os.path.join(self.test_dir, "success.c")
+        build_log = [{"directory": self.test_workspace,
+                      "command": "gcc -c " + source_file,
+                      "file": source_file
+                      }]
+
+        with open(build_json, 'w') as outfile:
+            json.dump(build_log, outfile)
+
+        print(analyze_cmd)
+        process = subprocess.Popen(
+                    analyze_cmd, stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE, cwd=self.test_dir)
+        out, err = process.communicate()
+        print(out+err)
+        errcode = process.returncode
+        self.assertEquals(errcode, 0)
+
+        # We expect the sucess stderr file in the success directory.
+        success_files = os.listdir(success_dir)
+        print(success_files)
+        self.assertEquals(len(success_files), 1)
+        self.assertIn("success.c", success_files[0])
+
```

Comment:
Please add an

```py
os.remove(os.path.join(success_dir, success_files[0]))
```

at the end. These tests should always go into an empty report dir.